### DISCUSSION TITLE
PHP 8.4: Implicitly marking parameters as nullable is deprecated

### DISCUSSION THREAD

**Discussion by jamieburchell** [deployphp/deployer#3911] [Bugs]
### Deployer Version

v7.4.0

### Target OS

Rocky Linux 9

### Which PHP version are you using?

PHP 8.3

### Steps to reproduce

PHP 8.4 will produce deprecation errors:

Implicitly marking parameter `$foo` as nullable is deprecated, the explicit nullable type must be used instead.

**Comment by antonmedv** [comment]
We have to support old 7.3 for now. In next Deployer release we will bump min php to 8.

**Reply by jamieburchell** [reply]
[Nullable types were introduced in 7.1](https://www.php.net/manual/en/migration71.new-features.php), so that's not a problem

> PHP supports nullable type declarations as of PHP 7.1 with the [?T syntax](https://wiki.php.net/rfc/nullable_types), and with the [T|null syntax](https://wiki.php.net/rfc/union_types_v2) as of PHP 8.0 when generalized support for union types was added.

[My PR](https://github.com/deployphp/deployer/pull/3910) fixes this.
